lib/pure/unicode.nim

```diff
@@ -20,9 +22,22 @@ type
 
 {.deprecated: [TRune: Rune, TRune16: Rune16].}
 
-proc `<=%`*(a, b: Rune): bool = return int(a) <=% int(b)
+proc `<=%`*(a, b: Rune): bool {.borrow.}
 proc `<%`*(a, b: Rune): bool = return int(a) <% int(b)
-proc `==`*(a, b: Rune): bool = return int(a) == int(b)
+proc `==`*(a, b: Rune): bool {.borrow.}
+
+proc `<`*(a, b: Rune): bool {.borrow.}
+proc `<=`*(a, b: Rune): bool {.borrow.}
+
+proc `<`*(a: Rune, b: RuneImpl): bool {.borrow.}
```

Comment:
Do we really need all these overloads? The point of a `distinct` type is to specifically __not__ allow things like this.